dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -242,7 +239,47 @@ internal class SdkFeature(
 
     // region Feature setup
 
-    private fun createStorage(
+    private fun prepareStorage(
+        dataUploadConfiguration: DataUploadConfiguration,
+        wrappedFeature: StorageBackedFeature,
+        context: Context,
+        persistenceStrategyFactory: PersistenceStrategy.Factory?
+    ): Storage {
+        val storageConfiguration = wrappedFeature.storageConfiguration
+        return if (persistenceStrategyFactory == null) {
+            val recentDelayMs = resolveBatchingDelay(coreFeature, storageConfiguration)
+            val filePersistenceConfig = coreFeature.buildFilePersistenceConfig().copy(
+                maxBatchSize = storageConfiguration.maxBatchSize,
+                maxItemSize = storageConfiguration.maxItemSize,
+                maxItemsPerBatch = storageConfiguration.maxItemsPerBatch,
+                oldFileThreshold = storageConfiguration.oldBatchThreshold,
+                recentDelayMs = recentDelayMs
+            )
+            setupMetricsDispatcher(dataUploadConfiguration, filePersistenceConfig, context)
+
+            createFileStorage(wrappedFeature.name, filePersistenceConfig)
+        } else {
+            createCustomStorage(wrappedFeature.name, storageConfiguration, persistenceStrategyFactory)
+        }
+    }
+
+    private fun createCustomStorage(
+        featureName: String,
+        storageConfiguration: FeatureStorageConfiguration,
+        persistenceStrategyFactory: PersistenceStrategy.Factory
+    ): Storage {
+        return AbstractStorage(
+            coreFeature.variant,
```

Comment:
it seems it should be `sdkCoreId` here rather than `coreFeature.variant`?

Reply:
Good catch indeed